### ISSUE TITLE
[BUDI-7803] Add support for OAuth authentication with REST APIs

### ISSUE THREAD

**Issue by pantherale0** [Budibase/budibase#5069]
**Describe the feature request**
Currently the only authentication options when using a REST API as a data source in Budibase is Basic and Bearer which makes using services such as Microsoft Graph to pull information in from an Azure Active Directory (for example) more complex to achieve as bearer tokens only last up to 1 hour so a middleware API would need to be created which uses its own bearer token/basic auth.

<sub>[BUDI-7803](https://linear.app/budibase/issue/BUDI-7803/add-support-for-oauth-authentication-with-rest-apis)</sub>

**Comment by 11harveyj** [comment]
Feature request

**Comment by 11harveyj** [comment]
Feature request

**Comment by christophsanz** [comment]
+1

**Comment by 11harveyj** [comment]
Feature request

**Comment by gilliganc** [comment]
hitting this issue as well

**Comment by mattkinnia** [comment]
+1 
Switched to a different service because the client credentials flow isn't supported.

**Comment by gilliganc** [comment]
> +1 Switched to a different service because the client credentials flow isn't supported.

yea we did as well

**Comment by LoKaDah** [comment]
This is desperately needed. The majority of other solutions have this already - shame to have to look elsewhere.

**Comment by systema365** [comment]
This is a shame, anyone know of a middleware application?  could oauth2-proxy and nginx work?

**Comment by christophe-lemoine** [comment]
Using Rest APIs without the option to have a dedicated and proper oAuth  per service is very limited.
We need to be able to define oAuth credentials per data source, with all oAuth options, and not just use the logged in user Bearer token.

**Comment by melohagan** [comment]
I agree @christophsanz - the issue must have been closed by accident.

**Comment by pantherale0** [comment]
I'm not using budibase at the moment due to this.

Would have reopened otherwise.

**Comment by pantherale0** [comment]
Why has this been closed with no feedback?

**Comment by shogunpurple** [comment]
@pantherale0 We are cleaning up our backlog of old issues that haven't had activity for a long time. Will re-open this one as still valid

**Comment by flub71** [comment]
How is this still not a feature after 3 years?

**Comment by flub71** [comment]
Whoops, this is now supported as of 3m ago. Close.
